French number token updates

ilissablech opened the pull request:
- Use French words for numbers and a few instances of plurals being used in rural areas of France.
- Properly set `spanBoundaries` parameter for multi-word tokens to be properly evaluated and matchied during linking.

aarthykc commented on tokens/fr.json:
```diff
@@ -623,9 +660,7 @@
     },
     {
         "tokens": [
-            "général",
```
Was removing this intentional?

ilissablech replied:
@aarthykc As everything is evaluated without accents these are not needed. Could happily keep them in. Does the carmen side need or benefit from the accents?

aarthykc reviewed (changes requested):
@ilissablech had questions about some of the tokens you added here. Could you add more details to the PR body about why some of these were added? Could you also touch upon why some of the accents were dropped for a couple of these replacements?

aarthykc commented on tokens/fr.json:
```diff
@@ -742,9 +780,9 @@
         "tokens": [
             "Mal",
-            "Maréchal"
+            "Marechal"
```
Why "Marechal" instead of "Maréchal"?

aarthykc commented on tokens/fr.json:
```diff
@@ -940,20 +984,36 @@
         ],
-        "full": "Rond-Point",
-        "canonical": "Rpt",
+        "full": "rond-point",
```
It shouldn't matter if it's capitalised or not from carmen's perspective, any reason why these were changed?

ilissablech commented:
> @ilissablech had questions about some of the tokens you added here. Could you add more details to the PR body about why some of these were added? Could you also touch upon why some of the accents were dropped for a couple of these replacements?

@aarthykc I reverted the instances where accents were dropped. I also removed the `passage commun` token, as we should handle that situation in pt2itp. Thanks for calling that out. Many other words were added in to aid in matching addresses to networks, found during our orphan analysis sprint. As we now added in functionality in pt2itp using spanBoundaries I updated a few of those token sets to include other commonly seen variations of tokens (ie. all the variations of `major general`.

aarthykc commented on tokens/fr.json:
```diff
@@ -781,21 +818,17 @@
         ],
         "full": "major general",
-        "canonical": "m gen",
+        "canonical": "mgen",
```
@ilissablech are we certain that the canonical representation for major general is mgen?

aarthykc commented on tokens/fr.json:
```diff
@@ -706,21 +742,22 @@
             "leiutenant general"
         ],
         "full": "leiutenant general",
-        "canonical": "l gen",
```
@ilissablech are we certain that the canonical representation for leiutenant general is lgen and not l gen?

aarthykc reviewed (approved):
👌🏽 thanks for working through the feedback @ilissablech!
